hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -4155,18 +4155,25 @@ public CompactionState getCompactionState(final TableName tableName) {
           continue;
         }
         RegionMetrics regionMetrics = sl.getRegionMetrics().get(regionInfo.getRegionName());
-        if (regionMetrics.getCompactionState() == CompactionState.MAJOR) {
-          if (compactionState == CompactionState.MINOR) {
-            compactionState = CompactionState.MAJOR_AND_MINOR;
-          } else {
-            compactionState = CompactionState.MAJOR;
-          }
-        } else if (regionMetrics.getCompactionState() == CompactionState.MINOR) {
-          if (compactionState == CompactionState.MAJOR) {
-            compactionState = CompactionState.MAJOR_AND_MINOR;
-          } else {
-            compactionState = CompactionState.MINOR;
+        if (regionMetrics != null) {
+          if (regionMetrics.getCompactionState() == CompactionState.MAJOR) {
+            if (compactionState == CompactionState.MINOR) {
+              compactionState = CompactionState.MAJOR_AND_MINOR;
+            } else {
+              compactionState = CompactionState.MAJOR;
+            }
+          } else if (regionMetrics.getCompactionState() == CompactionState.MINOR) {
+            if (compactionState == CompactionState.MAJOR) {
+              compactionState = CompactionState.MAJOR_AND_MINOR;
+            } else {
+              compactionState = CompactionState.MINOR;
+            }
           }
+        } else {
+          String getCompactionDetailsErrorMsg = String.format("Can not get compaction details for region: %s, it may be disabled or in transition.",
+            regionInfo.getRegionNameAsString());
+          LOG.error(getCompactionDetailsErrorMsg);
+          throw new Exception(getCompactionDetailsErrorMsg);
```

Comment:
I'm hesitant here for throwing a exception.   
Mybe we can just print a line of error logs and no exception.